src/test/java/com/chat/yourway/integration/controller/ChatControllerTest.java

```diff
@@ -0,0 +1,202 @@
+import com.chat.yourway.dto.request.MessagePublicRequestDto;
+import com.chat.yourway.dto.response.MessageResponseDto;
+import com.chat.yourway.integration.controller.websocketclient.TestStompFrameHandler;
+import com.chat.yourway.integration.extension.PostgresExtension;
+import com.chat.yourway.integration.extension.RedisExtension;
+import com.chat.yourway.model.token.Token;
+import com.chat.yourway.repository.TokenRedisRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.springtestdbunit.DbUnitTestExecutionListener;
+import com.github.springtestdbunit.annotation.DatabaseSetup;
+import java.util.concurrent.CompletableFuture;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener;
+import org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.messaging.simp.stomp.StompHeaders;
+import org.springframework.messaging.simp.stomp.StompSession;
+import org.springframework.messaging.simp.stomp.StompSessionHandlerAdapter;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import org.springframework.web.socket.WebSocketHttpHeaders;
+import org.springframework.web.socket.client.standard.StandardWebSocketClient;
+import org.springframework.web.socket.messaging.WebSocketStompClient;
+
+@Slf4j
+@ExtendWith({PostgresExtension.class, RedisExtension.class})
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestExecutionListeners({
+    DependencyInjectionTestExecutionListener.class,
+    DbUnitTestExecutionListener.class,
+    MockitoTestExecutionListener.class,
+    ResetMocksTestExecutionListener.class
+})
+@DatabaseSetup(value = {
+    "/dataset/mockdb/topic.xml",
+    "/dataset/mockdb/contact.xml",
+    "/dataset/mockdb/topic_subscriber.xml",
+    "/dataset/mockdb/message.xml"
+}, type = CLEAN_INSERT)
+public class ChatControllerTest {
+
+  @Autowired
+  private ObjectMapper objectMapper;
+  @Autowired
+  private TokenRedisRepository tokenRedisRepository;
+
+  @LocalServerPort
+  private int port;
+  private StompSession session;
+
+  //Test jwt access token for email = vasil@gmail.com, expiration = 19/09/2024
+  private final String JWT_ACCESS_TOKEN = "eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ2YXNpbEBnbWFpbC5jb20iLCJpYXQiOjE2OTc3MTQzNzYsImV4cCI6MTcyNjc0NDc3Nn0.d8cLceS967kh45h2P-cbQix5XAAo4LRYvGUZmo9HFHI";
```

Comment:
what will we do after 19/09/2024? probably should be some workaround that will help to generate access_token in runtime (also, u need to test some cases with expired token or not valid token)

Reply:
I can't replace the token during the ws session to test the case with token expiration